### PR TITLE
ROC-3719: Statement of means income

### PR DESCRIPTION
**Pull request by varma02**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-3719

### Change description
Update full admission - Statement of Means Income page

### CODE REVIEW

File: src/main/features/response/views/statement-of-means/monthly-income.njk [hunk 1]
```diff
@@ -70,14 +47,25 @@
 
         {% if form.model and form.model.canAddMoreRows() %}
           <div id="add-event-container" class="form-group">
-            {{ linkButton('Add another income', 'action[addRow]', 'button button-secondary') }}
+            {{ linkButton('Add another source of income', 'action[addRow]', 'button button-secondary') }}
           </div>
         {% endif %}
-      </div>
+      </div> #}
```
Comment (timja-kainos) [inline src/main/features/response/views/statement-of-means/monthly-income.njk]: ?

File: src/main/features/response/views/macro/statement-of-means/incomeExpenseSource.njk [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  <div class="form-group">
+    {{ radioGroup(
+      form = form,
+      label = label,
+      name = name,
+      labelBold = false,
+      options = [
+        {
+          label: 'Week',
```
Comment (robertparkinson) [inline src/main/features/response/views/macro/statement-of-means/incomeExpenseSource.njk]: Can you please add IncomeExpenseSchedule as a global variable in the nunjucks index.ts file and use it to output the value and display text? similar to how we use StatementType in offer - response.njk

File: src/main/features/response/views/statement-of-means/monthly-income.njk [hunk 3]
```diff
@@ -70,14 +52,24 @@
...
+      {{ saveAndContinueButton() }}  
+      
     </form>
   </div>
 </div>
+
 {% endblock %}
+
+{% macro incomeSource(form, name, label, hint) %}
+  {% set capitalisedName = name.slice(0, 1) | upper + name.slice(1) %}
```
Comment (robertparkinson) [inline src/main/features/response/views/statement-of-means/monthly-income.njk]: nunjucks might be a cleaner way to capitalise the name: 
https://mozilla.github.io/nunjucks/templating.html#capitalize

File: src/main/app/claims/responseModelConverter.ts [hunk 4]
```diff
@@ -260,4 +264,102 @@ export class ResponseModelConverter {
...
+        amountReceived: income.universalCreditSource.amount,
+        otherSource: undefined
+      })
+    }
+    if (income.jobseekerAllowanceIncomeSource && income.jobseekerAllowanceIncomeSource.populated) {
+      incomes.push({
+        type: IncomeType.JOB_SEEKERS_ALLOWANCE_INCOME_BASES,
+        frequency: income.jobseekerAllowanceIncomeSource.schedule.value as PaymentFrequency,
+        amountReceived: income.jobseekerAllowanceIncomeSource.amount,
+        otherSource: undefined
```
Comment (timja-kainos) [inline src/main/app/claims/responseModelConverter.ts]: otherSource should be optional in the types and not required to be put here if its always undefined,

File: src/main/app/common/calculate-monthly-income-expense/incomeExpenseSources.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    if (monthlyIncome.councilTaxSupportSource && monthlyIncome.councilTaxSupportSource.populated) {
+      incomeExpenseSources.push(IncomeExpenseSource.fromFormModel(monthlyIncome.councilTaxSupportSource))
+    }
+
+    if (monthlyIncome.pensionSource && monthlyIncome.pensionSource.populated) {
+      incomeExpenseSources.push(IncomeExpenseSource.fromFormModel(monthlyIncome.pensionSource))
+    }
+
+    if (monthlyIncome.anyOtherIncomePopulated) {
+      monthlyIncome.otherSources
```
Comment (timja-kainos) [inline src/main/app/common/calculate-monthly-income-expense/incomeExpenseSources.ts]: should be able to do `incomeExpenseSources.push(... result)`

File: src/main/features/response/routes/statement-of-means/monthly-income.ts [hunk 6]
```diff
@@ -13,21 +13,64 @@ import { User } from 'idam/user'
...
+  }
+
+  return CalculateMonthlyIncomeExpense.calculateTotalAmount(
+    incomeExpenseSources.incomeExpenseSources
+  )
+}
+
+function isValid (incomeExpenseSources: IncomeExpenseSources): boolean {
+  const validator = new Validator()
+  const errors = validator.validateSync(incomeExpenseSources)
```
Comment (timja-kainos) [inline src/main/features/response/routes/statement-of-means/monthly-income.ts]: I think this is simpler:
`return validator.validateSync(incomeExpenseSources).length === 0`

File: src/main/features/response/views/statement-of-means/monthly-income.njk [hunk 7]
```diff
@@ -1,83 +1,85 @@
...
 
-    {% if amountError %}<span id="{{ amountKey }}[label]" class="error-message">{{ t(amountError) }}</span>{% endif %}
-    {% if descriptionError %}<span id="{{ descriptionKey }}[label]" class="error-message">{{ t(descriptionError) }}</span>{% endif %}
+      {{ linkButton('Recalculate total monthly income', 'action[calculateMonthlyIncome]', 'button button-secondary calculate-montly-income-expense') }}
 
-    <label class="form-label-bold mobile-show" for="{{ amountKey }}">{{ index + 1 }} {{ t('. Name of income') }}</label>
-    {{ textInputInline(name = descriptionKey, form = form, inputClass='column-1', ariaLabelledBy = 'description-label') }}
-    {{ poundInputInline(name = amountKey, label = (index + 1) + '. How much?', form = form, ariaLabelledBy = 'amount-label') }}
+      <p class="heading-small calculation-outcome-container">
+        Total monthly income: £<span class="total-monthly-income-expense"> {{ totalMonthlyIncomeExpense | default('0.00') }} </span>
```
Comment (timja-kainos) [inline src/main/features/response/views/statement-of-means/monthly-income.njk]: missing translation

File: src/main/public/js/expandable-checkbox-option.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
+$(document).ready(function () {
```
Comment (timja-kainos) [inline src/main/public/js/expandable-checkbox-option.js]: in future I don't think we should write new code in jquery, govuk-frontend doesn't require it and all the browsers that you might need it for are out of support

**Pull request merged**